File: Programmers/17682/Solution.java
import java.util.Stack;

class Solution {
    public int solution(String dartResult) {
        int answer = 0;

        int sLen = dartResult.length();
        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < sLen; i++) {
            String str = dartResult.substring(i, i + 1);

            int num = 0;

            try {
                num = Integer.parseInt(str);

                if (!stack.isEmpty() && stack.peek() == 1 && num == 0) {
                    stack.pop();
                    stack.add(10);
                } else {
                    stack.add(num);
                }

            } catch (Exception e) {
                int n = 0;
                int m = 0;
                switch (str) {
                    case "S":
                        break;
                    case "D":
                        n = stack.pop();
                        stack.add(n * n);
                        break;
                    case "T":
                        n = stack.pop();
                        stack.add(n * n * n);
                        break;
                    case "*":
                        n = stack.pop();
                        if (!stack.isEmpty()) {
                            m = stack.pop();
                            stack.add(m * 2);
                        }
                        stack.add(n * 2);
                        break;
                    case "#":
                        n = stack.pop();
                        stack.add(n * -1);
                        break;
                }
            }

        }

        while (!stack.isEmpty()) {
            answer += stack.pop();
        }

        return answer;
    }
}

File: Programmers/17682/solution.py
def solution(dartResult):
    answer = list(dartResult)

    result = []

    for c, i in enumerate(answer):
        if i.isdigit():
            if int(i) == 0 and len(result) > 0 and result[-1] == 1 and dartResult[c - 1] == '1':
                result[-1] = 10
                continue
            else:
                result.append(int(i))
                continue

        if i == 'S':
            result[-1] = result[-1]
            continue
        elif i == 'D':
            result[-1] = result[-1] ** 2
            continue
        elif i == 'T':
            result[-1] = result[-1] ** 3
            continue
        elif i == '#':
            result[-1] = result[-1] * -1
            continue
        elif i == '*':
            result[-1] = result[-1] * 2
            if len(result) > 1:
                result[-2] = result[-2] * 2

    return sum(result)
